llvm/lib/CheerpUtils/PartialExecuter.cpp

```diff
@@ -0,0 +1,1393 @@
+				{
+					--i;
+					auto *Successor = i->getCaseSuccessor();
+					if (Successor == to) {
+						SI->removeCase(i);
+					}
+				}
+
+				if (SI->getDefaultDest() == to)
+				{
+					if (SI->getNumSuccessors() == 1)
+					{
+						SI->eraseFromParent();
+						new UnreachableInst(from->getParent()->getParent()->getContext(), from);
+						return;
+					}
+					
+					BasicBlock* another = nullptr;
+					for (SwitchInst::CaseIt i = SI->case_end(), e = SI->case_begin(); i != e;)
+					{
+						--i;
+						auto *Successor = i->getCaseSuccessor();
+						another = Successor;
+						SI->removeCase(i);
+						break;
+					}
+					assert(another);
+					SI->setDefaultDest(another);
+				}
+				return;
+			}
+			case Instruction::Invoke:
+			{
+				//Given that we currently do not treat invokes the only way to get here would be via having the Invoke itself being unreachable
+				BasicBlock* BB = I->getParent();
+				I->replaceAllUsesWith(UndefValue::get(I->getType()));
+				I->eraseFromParent();
+				new UnreachableInst(BB->getParent()->getParent()->getContext(), BB);
+				return;
+			}
+			case Instruction::Unreachable:
+			{
+				return;
+			}
+			default:
+			{
+				llvm::errs() << "Warning: " << *I << " not hanlded in removeEdgeBetweenBlocks\n";
+			}
+		}
+	}
+}
+	
+template <PartialInterpreter::VisitingPolicy policy>
+const llvm::BasicBlock* PartialInterpreter::visitBasicBlock(const llvm::BasicBlock* BB, const llvm::BasicBlock* from)
+{
+	assert(BB);
+	using namespace llvm;
+
+	ExecutionContext& executionContext = getLastStack();
+	executionContext.CurBB = const_cast<llvm::BasicBlock*>(BB);
+	executionContext.CurInst = executionContext.CurBB->begin();
+
+	while (getLastStack().CurInst != BB->end())
+	{
+		visitOuter<policy>(*getLastStack().CurInst++);
+	}
+
+	const Instruction* Term = BB->getTerminator();
+	const BasicBlock* next = nullptr;
+	
+	if (const BranchInst* BI = dyn_cast<BranchInst>(Term))
+	{
+		if (BI->isConditional())
+		{
+			if (isValueComputed(BI->getCondition()) && (getBitMask(BI->getCondition()) == BitMask::ALL) )
+			{
+				GenericValue V = getOperandValueExplicit(BI->getCondition(), executionContext);
+				if (V.IntVal == 0u)
+					next = BI->getSuccessor(1);
+				else
+					next = BI->getSuccessor(0);
+			}
+		}
+		else
+			next = BI->getSuccessor(0);
+	}
+	else if (const SwitchInst *SI = dyn_cast<SwitchInst>(Term))
+	{
+		if (isValueComputed(SI->getCondition()) && (getBitMask(SI->getCondition()) == BitMask::ALL) )
+		{
+			auto c = SI->findCaseValue(ConstantInt::get(SI->getFunction()->getParent()->getContext(), getOperandValueExplicit(SI->getCondition(), executionContext).IntVal ));
+			next = c->getCaseSuccessor();
+		}
+	}
+	else
+	{
+		assert(next == nullptr);
+		//Non handled terminator are fine, since next being nullptr means no information will be propagated
+	}
+
+	//TODO: RAII to pop
+	return next;
+}
+
+class ModuleData
+{
+	std::string error;
```

Comment:
This is never actually used

Reply:
This is passes (as reference) when creating the Interpreter. it might be used to report some recoverable errors (that we are assuming are not happening since we filter to only a limited set)